### PR TITLE
GG-39533 Fixed handling of empty and null network interface patterns

### PR DESCRIPTION
**Pull request by sk0x50**
https://ggsystems.atlassian.net/browse/GG-39533

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/spi/communication/tcp/internal/BlacklistFilter.java [hunk 1]
```diff
@@ -42,8 +42,12 @@ public class BlacklistFilter implements IgnitePredicate<InetAddress> {
     public BlacklistFilter(Collection<String> blacklist) {
         matchers = new ArrayList<>(blacklist.size());
 
-        for (String pattern : blacklist)
+        for (String pattern : blacklist) {
+            if (pattern == null || pattern.isEmpty())
```
Comment (solveme) [inline modules/core/src/main/java/org/apache/ignite/spi/communication/tcp/internal/BlacklistFilter.java:46]: Let's also verify that `pattern.trim()` is also not empty. Maybe we have a shortcut for this in `IgniteUtils`, so eventually we can write something like:
`if (isNotEmpty(pattern))
    matchers.add(new IPv4Matcher(pattern));`

File: modules/core/src/test/java/org/apache/ignite/spi/communication/tcp/internal/BlacklistNetwokInterfacesTest.java [hunk 2]
```diff
@@ -89,4 +89,18 @@ public void testBlacklistFilter() throws Exception {
             IgniteUtils.resetCachedLocalAddressAndHostNames();
         }
     }
+
```
Comment (solveme) [inline modules/core/src/test/java/org/apache/ignite/spi/communication/tcp/internal/BlacklistNetwokInterfacesTest.java:92]: let's add test-case for string like "    " (one or more whitespaces)

**Review by solveme — CHANGES_REQUESTED** [review]
Please take a look to my comments. WDYT?

File: modules/core/src/main/java/org/apache/ignite/spi/communication/tcp/internal/BlacklistFilter.java (hunk 1 above)
Comment (sk0x50) [reply modules/core/src/main/java/org/apache/ignite/spi/communication/tcp/internal/BlacklistFilter.java]: nice catch! fixed

File: modules/core/src/test/java/org/apache/ignite/spi/communication/tcp/internal/BlacklistNetwokInterfacesTest.java (hunk 2 above)
Comment (sk0x50) [reply modules/core/src/test/java/org/apache/ignite/spi/communication/tcp/internal/BlacklistNetwokInterfacesTest.java:94]: test added

**Review by solveme — COMMENTED** [review]
LGTM!

**Pull request merged**